### PR TITLE
Fix bugs in UG

### PR DESCRIPTION
**Pull request by donwong2308**
Fixed bugs and added acknowledgements for code reuse
- Properly define valid input for name 
    - #219 #203
- Properly define valid input for dorm tags 
    - #218
- Behaviour of Find Feature does not match expected behaviour 
    - #183 #173
- Find Command has vague or problematic phrasing: Command syntax KEYWORD is vague 
    - #177

### CODE REVIEW

**Review by nyankoclaws — APPROVED** [review]
LGTM

**Review by JustWeiHao — APPROVED** [review]
LGTM

**Review by larainezo — COMMENTED** [review]
LGTM

File: docs/UserGuide.md [hunk 1]
```diff
@@ -176,7 +176,7 @@ Edits the specified fields of an existing person in Dormie.
 - Edit function will replace the specified field with the new input
 - Edits the person at the specified INDEX. The index refers to the index number shown in the displayed person list. The index must be a positive integer. eg. 1, 2, 3
 - Existing values will be updated to the input values.
-- Minimum 1 field must be specified. Eg. `n/NAME` is a field.
+- Minimum 1 field must be specified. eg. `n/NAME` is a field.
```
Comment (PallonCX) [inline docs/UserGuide.md:179]: Should it be "E.g." instead? Since we use a period before it, it should be the start of a new sentence.

**Review by PallonCX — APPROVED** [review]
LGTM, just some small issues written in comment. Also, perhaps we should write a disclaimer that command summary is only for quick reference, and the actual behaviour should refer to the feature part itself.

File: docs/UserGuide.md (hunk 1 above)
Comment (donwong2308) [reply docs/UserGuide.md:179]: I standardized it to just eg. for all of it ah, looks more cohesive I guess haha

**Review by larainezo — APPROVED** [review]
LGTM

**Pull request merged**